typescript/packages/subsurface-viewer/src/components/Map.tsx

```diff
@@ -579,6 +592,28 @@ const Map: React.FC<MapProps> = ({
     lights,
     triggerResetMultipleWells,
 }: MapProps) => {
+    // Detect user rotating/panning.
+    const [isRotState, setIsRotState] = useState<boolean>(false);
+    const [isRotating, setIsRotating] = useState<boolean>(false);
+
+    React.useEffect(() => {
+        ZScaleOrbitController.setIsRotatingReference(setIsRotating);
+    }, [setIsRotating]);
+
+    useEffect(() => {
+        setIsRotState(isRotating);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isRotating]);
+
+    const layers_list = layers?.map((item) => {
```

Comment:
I don't think we should add this handling in the Map component. It should be fairly straightforward to use [`onDragstart`](https://deck.gl/docs/api-reference/core/deck#ondragstart), etc. to manipulate layers' state.

Reply:
Done :-)